### PR TITLE
US255 - As a developer, I want the jail behavior encapsulated, so that rolling out, paying the fee, or using a card is configurable.

### PR DESCRIPTION
**Pull request by nicholasnastase**
Task#290: Draw a sate diagram for in jail lifecycle.
Task#286: Create StandardJailStrategy
Task#287: Update PlayerState to include jail counters (my first few commits here were late last night, accidentally started implementing rules in Player.cs. Corrected that code and moved all rule-base/operation functionality to the proper StandardJailStrategy.cs file to stay on track with the current architecture).
Task#289: Create tests for forced pay, card use, and doubles escape
Task#288: Fire JailStateChanged Event

### CODE REVIEW

File: Assets/Scripts/Events/EventChannelTypes/JailStateChangedEventChannel.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (kescob0825) [inline Assets/Scripts/Events/EventChannelTypes/JailStateChangedEventChannel.cs]: It looks like these imports are not being used and should be removed.

File: Assets/Scripts/Events/EventDataStructures/JailStateChangedEvent.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
```
Comment (kescob0825) [inline Assets/Scripts/Events/EventDataStructures/JailStateChangedEvent.cs]: None of these imports are being used.

File: Assets/Scripts/Managers/Jail/StandardJailStrategy.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (kescob0825) [inline Assets/Scripts/Managers/Jail/StandardJailStrategy.cs]: These are not being used.

File: Assets/Scripts/Managers/Jail/IJailStrategy.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (kescob0825) [inline Assets/Scripts/Managers/Jail/IJailStrategy.cs]: These imports are not being used.

File: Assets/Tests/EditMode/JailTests/JailStrategyTests.cs [hunk 5]
```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (kescob0825) [inline Assets/Tests/EditMode/JailTests/JailStrategyTests.cs]: These are not being used.

File: Assets/Scripts/Events/EventChannelTypes/JailStateChangedEventChannel.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+/// </para>
+/// <para>
+/// Payload: <c>JailStateChangedEvent</c>
+/// </para>
+/// </summary>
+
+namespace Assets.Scripts.Events.EventChannelTypes
+{
+    [CreateAssetMenu(fileName = "JailStateChangedEventChannel", menuName = "Events/Jail State Changed Event Channel")]
+    public class JailStateChangedEventChannel : ScriptableObject
```
Comment (kescob0825) [inline Assets/Scripts/Events/EventChannelTypes/JailStateChangedEventChannel.cs]: I would recommend changing this to inherit from the EventChannel<T> that we already implemented. It's already a ScriptableObject and it will allow us to generalize it for future components.

**Review by kescob0825 — CHANGES_REQUESTED** [review]
# Code Review Checklist

**Completed for**: US #255 and Tasks #286 to #290 
**Code Review Completed By**: Karl Escobar

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [ ] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [ ] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Review by hdathert — APPROVED** [review]
# Code Review Checklist

**Completed for**: US #255 and Tasks #286 to #290  
**Code Review Completed By**: @hdathert 

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*
